### ISSUE TITLE
[BUG] HAR data not matching requests actual network traffic

### ISSUE THREAD

**Issue by smckee-r7** [microsoft/playwright#17745]
**Context:**
- Playwright Version: 1.24.2
- Operating System: Mac
- Node.js version: 16.16.0
- Browser: All

**Code Snippet**
```javascript
const { chromium } = require('playwright');

(async () => {
  const browser = await chromium.launch({
    headless: false
  });
  const context = await browser.newContext({ recordHar: { path: './harFile.har' } });
  const page = await context.newPage();

  await page.route('**/*', async (route) => {
    const headers = {
      ...route.request().headers(),
      ...{ 'injected-header': '1' }
    };
    route.continue({ headers });
  });

  await page.goto('https://www.google.com/');

  page.close();
  context.close();
  browser.close();
})();
```

**Describe the bug**
The har data that's recorded in `harFile.har` matches the original request information(before page.route() injects a header) rather than the actual network request traffic and therefore doesn’t contain the injected header. I would have thought that the HAR data should match the actual network requests and what the server would see?

Another example would be, when using page.route to change the request method type from GET to POST. Playwrights HAR file would incorrectly say it was a GET request when the server actually received a POST request.

**Comment by arlendarcy** [comment]
Additionally `OPTIONS` requests are missing from the output har file but if you output the har from the browser it contains the options requests.

**Comment by yury-s** [comment]
It was fixed by https://github.com/microsoft/playwright/pull/17027, you should update to Playwright v1.26.0+

**Comment by arlendarcy** [comment]
> It was fixed by #17027, you should update to Playwright v1.26.0+
@yury-s `OPTIONS` requests are still missing on v1.26.1

**Comment by smckee-r7** [comment]
@yury-s that's perfect, thanks for that! I have found & opened another related issue that doesn't appear to be fixed yet #17802
